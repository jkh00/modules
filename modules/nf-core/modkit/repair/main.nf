process MODKIT_REPAIR {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ont-modkit:0.5.0--hcdda2d0_1':
        'biocontainers/ont-modkit:0.5.0--hcdda2d0_1' }"

    input:
    tuple val(meta), path(before_trim), path(after_trim)

    output:
    tuple val(meta), path("*.bam"), emit: bam
    tuple val(meta), path("*.log"), emit: log
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    modkit \\
        repair \\
        $args \\
        --threads ${task.cpus} \\
        --donor-bam ${before_trim} \\
        --acceptor-bam ${after_trim} \\
        --output-bam ${prefix}_repaired.bam \\
        --log-filepath ./${prefix}_repair.log


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        modkit: \$(modkit --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"


    """
    echo $args
    
    touch ${prefix}.bam
    touch ${prefix}.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        modkit: \$(modkit --version)
    END_VERSIONS
    """
}
