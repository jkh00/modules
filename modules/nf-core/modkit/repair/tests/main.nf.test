nextflow_process {

    name "Test Process MODKIT_REPAIR"
    script "../main.nf"
    process "MODKIT_REPAIR"

    tag "modules"
    tag "modules_nfcore"
    tag "modkit"
    tag "modkit/repair"

    test("test - donor_bam - acceptor_bam") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("https://github.com/nf-core/test-datasets/raw/refs/heads/methylong/test_data/test_modbam/ont_Col-0_tets_data_Chr1_5mb_7mb.bam", checkIfExists: true),
                    file("https://github.com/nf-core/test-datasets/raw/refs/heads/methylong/test_data/test_modbam/trimmed_ont_Col-0_tets_data_Chr1_5mb_7mb.bam", checkIfExists: true),
                ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.bam.collect { tuple -> file(tuple[1]).name },
                    process.out.log.collect { tuple -> file(tuple[1]).name },
                    process.out.versions.collect { tuple -> file(tuple[1]).name }
                ).match() }
            )
        }

    }

    test("test - donor_bam - acceptor_bam - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("https://github.com/nf-core/test-datasets/raw/refs/heads/methylong/test_data/test_modbam/ont_Col-0_tets_data_Chr1_5mb_7mb.bam", checkIfExists: true),
                    file("https://github.com/nf-core/test-datasets/raw/refs/heads/methylong/test_data/test_modbam/trimmed_ont_Col-0_tets_data_Chr1_5mb_7mb.bam", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
