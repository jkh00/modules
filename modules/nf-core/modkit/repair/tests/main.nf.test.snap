{
    "test - donor_bam - acceptor_bam": {
        "content": [
            [
                "test_repaired.bam"
            ],
            [
                "test_repair.log"
            ],
            [
                "d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-09-16T11:40:17.927022229"
    },
    "test - donor_bam - acceptor_bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,68970b3d8073277b0f988c459c694e9c"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68970b3d8073277b0f988c459c694e9c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-09-16T11:40:29.802588248"
    }
}